extensions/2.0/Vendor/EXT_mesh_features/README.md

```diff
@@ -616,13 +617,69 @@ Texture filtering must be `9728` (NEAREST), `9729` (LINEAR), or undefined, for a
+| ...                      | ...                      | ...             | ...                    |
+| `"UINT32"`               | `5125` <sup>2</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"UINT64"`               | `5121` <sup>3</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"INT8"`                 | `5120` (BYTE)            | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"INT16"`                | `5122` (SHORT)           | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"INT32"`                | `5124` (INT)<sup>4</sup> | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"INT64"`                | `5121` <sup>3</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"FLOAT32"`              | `5126` (FLOAT)           | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"FLOAT64"`              | `5121` <sup>3</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| ...                      | ...                      | ...             | ...                    |
+| `"BOOLEAN"`              | `5121` <sup>3</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| `"BOOLEAN"`              | `5121` <sup>3</sup>      | `"ARRAY"`       | `"SCALAR"`<sup>1</sup> |
+| `"STRING"`               | `5121` <sup>3</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| `"STRING"`               | `5121` <sup>3</sup>      | `"ARRAY"`       | `"SCALAR"`<sup>1</sup> |
+| `"ENUM"`                 | `5121` <sup>3</sup>      | `"SINGLE"`      | `"SCALAR"`             |
+| `"ENUM"`                 | `5121` <sup>3</sup>      | `"ARRAY"`       | `"SCALAR"`<sup>1</sup> |
+
+<small><sup>1</sup> Use "SCALAR" accessors for "ARRAY" property types, with `componentCount` for fixed-length arrays.</small><br>
+<small><sup>3</sup> `5125` (UNSIGNED_INT) disallowed by base glTF 2.0 specification except as `primitive.indices`. `EXT_mesh_features` overrides this restriction.</small><br>
+<small><sup>3</sup> `5121` (UNSIGNED_BYTE) used as general-purpose binary storage.</small><br>
+<small><sup>4</sup> `5124` (INT) disallowed by base glTF 2.0 specification. `EXT_mesh_features` overrides this restriction.</small>
+
+Each accessor `byteOffset` must be aligned to 4-byte boundaries, or to 8-byte boundaries for 64-bit property types.
```

Comment:
Is there an inherent glTF restriction for the 4-byte alignment requirement?

Reply:
Does it have to do with GPU attribute alignment requirements since these are accessors?

Reply:
From [3.6.2.4. Data Alignment](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#data-alignment) —

> The offset of an accessor into a bufferView (i.e., accessor.byteOffset) and the offset of an accessor into a buffer (i.e., accessor.byteOffset + bufferView.byteOffset) MUST be a multiple of the size of the accessor’s component type.

> ... each element of a vertex attribute MUST be aligned to 4-byte boundaries inside a bufferView (i.e., accessor.byteOffset and bufferView.byteStride MUST be multiples of 4).

I didn't go as far as requiring every _element_ of a property values accessor to be aligned to 4-byte boundaries — that's lots of storage overhead, with no guarantee the client will use it as a vertex attribute. But I did want to specify alignment for `accessor.byteOffset`, since that affects more than just vertex attributes and has little overhead.

Requiring 4-byte or 8-byte accessor alignment is conservative, and 1-byte or 2-byte types could do with less. But this seemed simpler than using generic language like "the size of the value". For scalar numeric types like `int8|16|32`, `uint8|16|32`, and `float32` the size of the value is trivial, but for other types it's more complicated:

1. `boolean` and `string` types we need alignment to accessor component size, not value size.
2. `int64` and `uint64` types we need alignment to value size, not accessor component size.<sup>1</sup>
3. vector, matrix, and array types we need alignment to component size derived from rules 1–2 above.

<small><sup>1</sup> Based on your earlier comment I see that `uint64` and `int64` do indeed have GL enums, so we could keep the accessor size aligned here. But there may be other 64-bit property types we want to support in the future, which would map to `uint8` storage, in which case accessor component size is too small.</small>